src/common.rs

```diff
@@ -126,6 +153,56 @@ pub enum TableKey {
     Key(String, Vec<Column>),
 }
 
+impl fmt::Display for TableKey {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self{
+            TableKey::PrimaryKey(ref columns) => {
+                write!(f, "PRIMARY KEY (")?;
+                for (i, column) in columns.iter().enumerate(){
+                    if i > 0 { write!(f, ", ")?; }
+                    write!(f, "{}", column)?;
+                }
```

Comment:
(missing a `, "{}", ` there)

Or for that matter, just use `write!(f, "{:?}", columns)`

Reply:
`{:?}` is no good because it will display the columns as `Column { ... }` (and ought to). I've fixed the above code though -- it was also missing the `map(|c| c.name)` part.

Reply:
Had to add `to_owned()` to `c.name` to satisfy the compiler from error: `^ cannot move out of borrowed content`
```rust
write!(f, "{}", columns.iter().map(|c| c.name.to_owned()).collect::<Vec<_>>().join(", "))
```